make/Makefile.common

```diff
@@ -161,12 +162,12 @@ src-symlinks:
 	@$(foreach f, $(GEM5_SRCS), cp -asf $(ALADDIN_HOME)/$(f) $(BUILD_DIR)/gem5/$(notdir $(f));)
 	@mkdir -p $(BUILD_DIR)/bin
 
-protos: $(BUILD_PROTO_CPP_SRCS) $(PROTO_PY_SRCS)
+protos: $(BUILD_PROTO_CPP_SRCS) $(BUILD_PROTO_PY_SRCS)
 
 $(BUILD_DIR)/%.pb.cpp: %.proto
 	$(PROTOC) -I=$(SMAUG_HOME) --cpp_out=. $^
 	mv $(basename $^).pb.cc $@
 	mv $(basename $^).pb.h $(dir $@)
 
-%_pb2.py: %.proto
-	$(PROTOC) -I=$(SMAUG_HOME) --python_out=. $^
+$(BUILD_DIR)/%_pb2.py: %.proto
+	$(PROTOC) -I=$(SMAUG_HOME) --python_out=$(BUILD_DIR) $^
```

Comment:
This fails because the Python files import these protos using `import smaug.core.xxx_pb2` and now they are generated in the `build` folder, which is not in Pyhton's search path (we only add `SMAUG_HOME` to the search path).

Reply:
Oops. Fixed.

Reply:
If we don't add `build` into the Python search path (which is strange to do), the imports of the Python protos will still fail. That's why I generated these protos directly under the source folder.

Reply:
Autogenerated code is used all the time (just look at gem5's ISA definitions/SLICC), but the outputs are not put in the source tree because they're not, well, original source, and it defeats the purpose of isolating build products from source code, which is generally a Good Thing. The typical thing to do in this case would be to package SMAUG and `pip install` it, which would put everything (human-written source code and proto-generated code) in a single location. I actually did consider making a pip package for SMAUG, but since we're expecting people to be actively developing on top of it, that would probably introduce more headaches than it would resolve.

I agree that it's unconventional to put `build` into `PYTHONPATH`, but that's only because it's not the *final* destination of the outputs. We could easily set up a convention in which the user is expected to `make install` afterwards and add that path to `PYTHONPATH` instead.